airflow/providers/microsoft/azure/hooks/adls.py

```diff
@@ -0,0 +1,221 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from typing import Any
+
+from azure.core.exceptions import ResourceExistsError, ResourceNotFoundError
+from azure.identity import ClientSecretCredential
+from azure.storage.filedatalake import (
+    DataLakeDirectoryClient,
+    DataLakeFileClient,
+    DataLakeServiceClient,
+    FileSystemClient,
+)
+
+from airflow.hooks.base import BaseHook
+
+
+class AzureDataLakeStorageV2(BaseHook):
```

Comment:
Now I changed the Hook name now.

Reply:
I'm not sure about this.
eventually v2 will be the standard and on that day it would be redundant to mention v2.

I see that all current v1 services are listed as `AzueDataLakeStorage*`
WDYT about having the new ones as `Adls*`?

Reply:
Yes, the current  v1 services are listed as AzueDataLakeStorage. Now it is `AzureDataLakeStorageClient` can it be `AzureDataLakeStorageClientHook`. WDYT?

Reply:
since we don't have hook for adls it can be anything we want.
i suggested the pattern: `Adls*` so:


| Current V1         | Suggested V2       |
|--------------------|--------------------|
|          -         | AdlsClientHook     |
| ADLSDeleteOperator | AdlsDeleteOperator |
| ADLSListOperator   | AdlsListOperator   |